Libplanet.Explorer/GraphTypes/BlocksQuery.cs

```diff
@@ -116,6 +153,56 @@ public BlocksQuery(BlockChain<T> chain)
             }
         }
 
+        private IEnumerable<Transaction<T>> ListTransactions(
+            Address? signer, Address? involved, bool desc, long offset, int? limit)
+        {
+            Block<T> tip = _chain.Tip;
+            long tipIndex = tip.Index;
+
+            if (offset < 0)
+            {
+                offset = tipIndex + offset + 1;
+            }
+
+            if (tipIndex < offset || offset < 0)
+            {
+                yield break;
+            }
+
+            Block<T> block = desc ? _chain[tipIndex - offset] : _chain[offset];
+
+            while (limit is null || limit > 0)
+            {
+                int index = desc ? block.Transactions.Count() - 1 : 0;
+                while (index >= 0 && index < block.Transactions.Count())
+                {
+                    Transaction<T> tx = block.Transactions.ElementAt(index);
```

Comment:
As [`Enumerable.ElementAt()`][1] does not assume an object is randomly accessible but only sequentially, IMHO the following code would be more predictable, time-complexity-wise:

```suggestion
                    Transaction<T> tx = block.Transactions[index];
```

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.elementat?view=netcore-2.2

Reply:
Since `block.Transaction` is defined as `IEnumerable`, thus `block.Transactions[index]` is invalid. Should I use ToList() method or something?

Reply:
Hmm… I was confused that <https://github.com/planetarium/libplanet/pull/355> was already merged.  How about just using `foreach` instead of `while`?

Reply:
Hmm… I was confused that <https://github.com/planetarium/libplanet/pull/355> was already merged.  How about just using `foreach` instead of `while`?

Reply:
I guess `foreach` is inappropriate because of `desc` option...

Reply:
How about `foreach (var tx in desc ? block.Transactions.Reverse() : block.Transactions)`?

See also [`Enumerable.Reverse()` method][1].

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.reverse?view=netcore-2.2